feat: add filtering and log priority functionality to Android native logs

Doug821 opened the pull request:
# Description

This pull request adds filtering functionality to the Android native logs. The changes include the addition of a new `isIgnoredTag` method and an updated version of `readOutputLogs` that takes a `tags` parameter to filter out log lines that contain any of the specified tags.

In addition, the `setUpRedirectLogs` method also was updated and it can now receive an optional string parameter that specifies the log priority level.

### Changes Made

- Added `isIgnoredTag` method to check if a log line should be ignored based on the specified tags
- Updated `readOutputLogs` to take a `tags` parameter and filter out log lines that contain any of the specified tags
- Updated `setUpRedirectLogs` method to receive an optional log priority level parameter

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Result

With these changes, the Android native logs module now provides a more customizable logging experience, allowing developers to filter specific log tags and set log priority levels as needed.

Doug821 committed:
feat: Add filtering functionality to Android logs

This commit adds filtering functionality to the `readOutputLogs` method in the Android native logs module. The new version of the method now accepts an optional array of tags to ignore when reading logs from the file.

The commit also updates the `redirectLogs` method to accept a shortened version of the log level string instead, allowing also to add filters based on the log level priority.

Both changes are specific to Android devices and improve the functionality of the native logs module.

V1pi commented on android/src/main/java/com/reactlibrary/RNNativeLogsModule.java:
```diff
@@ -29,37 +30,52 @@ public String getName() {
 
     @ReactMethod
-    public void setUpRedirectLogs(String fileIdentifier, final Promise promise) {
+    public void setUpRedirectLogs(String fileIdentifier, String logLevel, final Promise promise) {
```
This a breaking change for iOS. You should keep the old signature as well (the one without the log level). iOS does not have the logLevel signature so will fail. Same for the others methods

V1pi commented on src/index.ts:
```diff
@@ -4,36 +4,39 @@ const { RNNativeLogs } = NativeModules;
   */
-  async getNewLogs(identifier: string): Promise<string[] | null> {
-    const allNativeLogs = await RNNativeLogs.readOutputLogs(identifier);
+  async getNewLogs(identifier: string, tags: string[]): Promise<string[] | null> {
```
Keep old signature as well

Doug821 committed:
chore: add method overloading for logging functionality

This commit adds method overloading for logging functionality. The overloading was introduced in two new methods: `setUpRedirectLogs` and `readOutputLogs`.

These methods provide the same functionality as their counterparts but with additional parameters that allow more control over the logging process.

This overloading was added to ensure backward compatibility with iOS apps and prevent breaking changes.

Doug821 committed:
chore: add method overloading and platform checks

V1pi commented on src/index.ts:
```diff
@@ -1,39 +1,53 @@
+
   /** Gets all logs from the given file.
     * @param {string} identifier - The name of the file to read the logs to.
+    * @param {string[]} tags - An array of tags to exclude from the logs. Only logs that do not match these tags will be redirected.
```
Add Android only comment

V1pi commented on src/index.ts:
```diff
@@ -1,39 +1,53 @@
   */
-  async getLogs(identifier: string): Promise<string[] | null> {
-    const allNativeLogs = await RNNativeLogs.readOutputLogs(identifier);
+  async getLogs(identifier: string, tags: string[]): Promise<string[] | null> {
```
Should be optional

V1pi commented on src/index.ts:
```diff
@@ -1,39 +1,53 @@
-  async getNewLogs(identifier: string): Promise<string[] | null> {
-    const allNativeLogs = await RNNativeLogs.readOutputLogs(identifier);
+  async getNewLogs(identifier: string, tags?: string[]): Promise<string[] | null> {
+    let allNativeLogs: string[];
```
don't use let, use const. You can do a ternary operation to check if it's undefined or iOS to call without tags. Otherwise call with tags

V1pi commented on src/index.ts:
```diff
@@ -1,39 +1,53 @@
     * @returns {Promise<void>} - A promise that resolves when the logs are redirected.
   */
-  async redirectLogs(identifier: string): Promise<void> {
+  async redirectLogs(identifier: string, logLevel?: string): Promise<void> {
```
If we can only accepts few parameters create a type called LogLevel and put only the possible values.
`type LogLevel = 'D' | 'I'`

Use it in the others place as well

Doug821 committed:
fix: add filtering tags as optional parameter for getLogs

Doug821 committed:
chore: add missing types

V1pi commented on src/index.ts:
```diff
@@ -1,39 +1,53 @@
+
   /** Redirects the native logs from the console to a given file.
     * @param {string} identifier - The name of the file to write the logs to.
+    * @param {LogLevel} logLevel - [ANDROID only] The log level to redirect. Can be one of: "D" for debug, "I" for info, "W" for warn, "E" for error, "F" for fatal.
```
Add "LogLevel | undefined"

V1pi commented on src/index.ts:
```diff
@@ -1,39 +1,53 @@
+  async redirectLogs(identifier: string, logLevel?: LogLevel): Promise<void> {
     this.currentLogsIndex[identifier] = 0;
-    await RNNativeLogs.setUpRedirectLogs(identifier);
+    if (Platform.OS === "ios" || !logLevel) {
```
Make it simpler

V1pi commented on src/index.ts:
```diff
@@ -1,39 +1,53 @@
-  async getLogs(identifier: string): Promise<string[] | null> {
-    const allNativeLogs = await RNNativeLogs.readOutputLogs(identifier);
+  async getLogs(identifier: string, tags?: string[]): Promise<string[] | null> {
+    const allNativeLogs = await RNNativeLogs.readOutputLogs(identifier, tags);
```
Add ternary condition

Doug821 committed:
chore: add optional parameters type description and simplify conditions.
